src/category/dto/subcategory.dto.ts

```diff
@@ -14,5 +14,7 @@ export class CreateSubCategoryDto {
   type: string;
 
   @IsOptional()
-  parentId?: number;
+  @IsNotEmpty({ message: 'Type cannot be empty' })
+  @IsInt({ message: 'Type must be a string' })
```

Comment:
Since the type of parentId is number the message should be "Type must be a number".